homeassistant/components/androidtv_remote/media_player.py

```diff
@@ -0,0 +1,185 @@
+"""Media player support for Android TV Remote."""
+from __future__ import annotations
+
+import asyncio
+from typing import Any
+
+from androidtvremote2 import AndroidTVRemote
+
+from homeassistant.components.media_player import (
+    MediaPlayerDeviceClass,
+    MediaPlayerEntity,
+    MediaPlayerEntityFeature,
+    MediaPlayerState,
+    MediaType,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import AndroidTVRemoteBaseEntity
+
+PARALLEL_UPDATES = 0
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Android TV media player entity based on a config entry."""
+    api: AndroidTVRemote = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities([AndroidTVRemoteMediaPlayerEntity(api, config_entry)])
+
+
+class AndroidTVRemoteMediaPlayerEntity(AndroidTVRemoteBaseEntity, MediaPlayerEntity):
+    """Android TV Remote Media Player Entity."""
+
+    _attr_assumed_state = True
```

Comment:
First of all, it's convenient. We don't need to create a custom card for basic playback control (play and pause, next/prev, stop and volume control). 

The media player card already contains everything you need to control it out of the box:

<img src="https://user-images.githubusercontent.com/11841379/235256286-c185e39d-5ffd-468a-a732-02dd65b1cca9.png">

I think users expects a media player when they add an integration that is now native to Android TV.

In addition, the `Google Cast` integration displays the status of the device as `off` when it is not playing anything or the Android TV app is not set playback info in `MediaSession`, even if the device is turned on. This integration closes this gap. By combining two media players through `universal` integration, we get almost complete control on the Android TV device. 

Combining two media players through Universal integration is very simple:

```yaml
media_player:
  - platform: universal
    name: My Android TV
    children:
      - media_player.android_tv_cast
      - media_player.android_tv_remote
```

The Universal integration will automatically select the active player based on state order: 
https://github.com/home-assistant/core/blob/65837c9075814709d40067aff64992d623b6f8f4/homeassistant/components/universal/media_player.py#L103-L113

Reply:
This may be a bit random, but I've been following this functionality a bit after discovering the original PR in my search of information on how to build this myself since the ADB integration felt unstable and slow.

I've been thinking about this comment for a few days and at first I agreed: it does not make sense to add a media_player entity if there is no real information about media playback being exposed by the remote service. And you can just create your own using the universal media player.

I am thus currently running the beta of 2023.5 and I have created a universal media player to be able to control my Shield TV using a single media player card.

However. Like mentioned by Drafteed it feels like the AndroidTV integration is meant to improve upon the ADB integration. And it should be perfectly capable of doing so as media_player entity as well, except for reporting the current playback state.

This big piece of YAML I have contains a service call for on/off, play/pause, volume changing etc. It works perfectly fine,  except for that _everyone_ who wants to use the Android TV integration for media player support needs to go this universal media player route (and based on the performance so far everyone probably wants this). Like mentioned in the original TV remote PR: ['UX counts'](https://github.com/home-assistant/core/pull/89935#issuecomment-1475839978). Having to create a YAML media player seems like bad UX for an integration that can support 90% of the media player features by just exposing it's own entity.

Reply:
In fact, many media players in popular Home Assistant integrations do not give playback status, for example: 

- [Sony Bravia TV](https://github.com/home-assistant/core/blob/cc4e741cfa21973cae47a0726e83ec2e51f8ba97/homeassistant/components/braviatv/media_player.py#L43)
- [LG WebOS TV](https://github.com/home-assistant/core/blob/cc4e741cfa21973cae47a0726e83ec2e51f8ba97/homeassistant/components/webostv/media_player.py#L123)
- [LG Netcast TV](https://github.com/home-assistant/core/blob/cc4e741cfa21973cae47a0726e83ec2e51f8ba97/homeassistant/components/lg_netcast/media_player.py#L87)
- [Phillips TV](https://github.com/home-assistant/core/blob/cc4e741cfa21973cae47a0726e83ec2e51f8ba97/homeassistant/components/philips_js/media_player.py#L487)
- [Panasonic Viera](https://github.com/home-assistant/core/blob/cc4e741cfa21973cae47a0726e83ec2e51f8ba97/homeassistant/components/panasonic_viera/__init__.py#L247-L270)
- [Samsung TV](https://github.com/home-assistant/core/blob/cc4e741cfa21973cae47a0726e83ec2e51f8ba97/homeassistant/components/samsungtv/media_player.py#L121-L122)
- [Xiaomi TV](https://github.com/home-assistant/core/blob/cc4e741cfa21973cae47a0726e83ec2e51f8ba97/homeassistant/components/xiaomi_tv/media_player.py#L61)

Moreover, without the `media_player` entity we cannot get the volume info from Remote Service.

Reply:
It seems most of the functionality in this PR, other than volume_set, could be added using a Universal Media Player. The Google Cast integration already supports volume_set so it doesn't sound that important.

If users, for better experience, have to use the Universal Media Player to combine the 2 media players anyway, they might as well copy paste a few more lines of code and avoid having this "duplicate" player with missing playback status.

To be clear, I'm not objecting to this PR. I'm trying to understand what this tries to address and what are the alternatives. At the very least, clarify in the documentation the limitations of this media player (lack of playback status and limited play_media support). It might also be good to provide an example how to combine this media player with the Google Cast one using Universal Media Player.

Reply:
> The Google Cast integration already supports volume_set so it doesn't sound that important.

Ok, let's play media through a player that doesn't support [MediaSession](https://developer.android.com/reference/android/media/session/MediaSession) (e.g. X-plore or MX Player):

<img src="https://user-images.githubusercontent.com/11841379/236493004-10a3ec68-ef70-4045-a37e-c3c91feed6a5.png">

As we can see, the Google Cast player is turned off and does not allow us to control playback, including volume level.

Now let's start Steam Link or some Android TV game:

<img src="https://user-images.githubusercontent.com/11841379/236494127-244435c6-dd4d-4f40-a649-96cd01ceb925.png">

Exactly the same situation, volume control in this case is also not available.

>  I'm trying to understand what this tries to address and what are the alternatives. 

I don't know how else to convince you that the `media_player` platform will be useful here. I have already described all the benefits in my posts above. Please also take a look [this stream record from the Home Assistant 2023.5 Release Party](https://youtu.be/ZPrlZxDE6Zk?t=5269) where the maintainers commented that adding a media player would be a good thing.

> At the very least, clarify in the documentation the limitations of this media player (lack of playback status and limited play_media support)

Have you seen [my documentation pull request](https://github.com/home-assistant/home-assistant.io/pull/27059)? 
I noted this in the documentation:

> Due to API limitations, the integration will not display the playback status. It is recommended to use this integration together with [Google Cast integration](https://www.home-assistant.io/integrations/cast/). Two media players can be combined into one using the [Universal Media Player](https://www.home-assistant.io/integrations/universal/) integration.